lib/solr/query/response/parser.rb

```diff
@@ -72,7 +72,8 @@ def parse_documents
         def parse_regular_documents
           solr_response['response']['docs'].map do |d|
             debug_info = solr_response.dig('debug', 'explain', d['id'])
-            Document.new(id: d['id'], score: d['score'], debug_info: debug_info)
+            fields = d.dup.delete_if { |k, v| %w[id score].include?(k) }
+            Document.new(id: d['id'], score: d['score'], debug_info: debug_info, fields: fields)
```

Comment:
What do you think about moving `id` and `score` to `fields` and add the following code to `Document`?
```ruby
class Document
  def id
    fields['id']
  end

  def score
    fields['score']
  end
end
```

Reply:
One important note: I made `id` a mandatory field, so that's one of the points why I decided to keep `id` as a separate attribute.